cmake/Modules/FindWebp_ep.cmake

```diff
@@ -0,0 +1,130 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# Fetches and builds libwebp for use by tiledb.
+# TBD...
+# This module defines:
+#   - libmagic_INCLUDE_DIR, directory containing headers
+#   - libmagic_LIBRARIES, the Magic library path
+#   - libmagic_FOUND, whether Magic has been found
+#   - The libmagic imported target
+
+# Include some common helper functions.
+include(TileDBCommon)
+
+# Search the path set during the superbuild for the EP.
+set(WEBP_PATHS ${TILEDB_EP_INSTALL_PREFIX})
+
+if (TILEDB_WEBP_EP_BUILT)
+  find_package(webp PATHS ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_DEPS_NO_DEFAULT_PATH})
+endif()
+
+if (TILEDB_WEBP_EP_BUILT)
+  find_path(webp_INCLUDE_DIR
+    NAMES encode.h decode.h demux.h mux.h mux_types.h
+    PATHS ${WEBP_PATHS}
+    PATH_SUFFIXES include/webp
+    ${NO_DEFAULT_PATH}
+  )
+
+  # Link statically if installed with the EP.
+  # TBD: Does this find/include the multiple libraries comprising webp?
+  find_library(webp_LIBRARIES
+    webp
+    PATHS ${WEBP_PATHS}
+    PATH_SUFFIXES lib a
+    #${TILEDB_DEPS_NO_DEFAULT_PATH}
+    ${NO_DEFAULT_PATH}
+  )
+  
+  message(STATUS "B4 webp_INCLUDE_DIR is ${wepb_INCLUDE_DIR}")
+  message(STATUS "B4 webp_LIBRARIES is ${wepb_LIBRARIES}")
+
+  include(FindPackageHandleStandardArgs)
+  FIND_PACKAGE_HANDLE_STANDARD_ARGS(webp
+    REQUIRED_VARS webp_LIBRARIES webp_INCLUDE_DIR
+  )
+
+  message(STATUS "AF webp_INCLUDE_DIR is ${wepb_INCLUDE_DIR}")
+  message(STATUS "AF webp_LIBRARIES is ${wepb_LIBRARIES}")
+
+endif()
+
+# if not yet built add it as an external project
+if(NOT TILEDB_WEBP_EP_BUILT)
+  if (TILEDB_SUPERBUILD)
+    message(STATUS "Adding Webp as an external project")
+
+    #if (WIN32)
+    #  set(CFLAGS_DEF "")
+    #else()
+    #  set(CFLAGS_DEF "${CMAKE_C_FLAGS} -fPIC")
+    #  set(CXXFLAGS_DEF "${CMAKE_CXX_FLAGS} -fPIC")
+    #endif()
+
+    # For both windows/nix using tiledb fork of file-windows from julian-r who has build a cmake version and patched for msvc++
+    # that was modified by tiledb to also build with cmake for nix
+    ExternalProject_Add(ep_webp
+      PREFIX "externals"
+      GIT_REPOSITORY "https://chromium.googlesource.com/webm/libwebp"

```

Comment:
Can we use tarball or zip download instead? This is generally preferable to git cloning for external projects.

Reply:
Two days ago, the 'static' build support in CMakeLists.txt was not yet in a release.

Reply:
Think static option was added to their branch 'main' apr 12, 2022.

Last prior (tag) release appears to have been v1.2.2 on Jan 20, 2022, tho the 1.2.2 -branch- appears as updated after on  apr 7 2022 (so changes release 'branch' changes 3mon-or-so after 'tag' release).
Prior (tag) release v1.2.1 dated Aug 13 2021 (Friday the 13th), with the 1.2.1 -branch- being (last) updated afterward (also) on Apr 7 2022.

So not sure when the static build option will make it into (tag) release.

***Is it still prefereable to fetch the .tgz (for 'main' branch specific commit) even tho there isn't yet a (tag) release?

Reply:
Changed to use commit out of 'main' branch as 'static' build support not found in any branch/tag yet.